rtos/mbed_boot.c

```diff
@@ -319,7 +320,10 @@ void mbed_start_main(void)
     _main_thread_attr.cb_mem = _main_obj;
     _main_thread_attr.priority = osPriorityNormal;
     _main_thread_attr.name = "MAIN";
-    osThreadNew((osThreadFunc_t)pre_main, NULL, &_main_thread_attr);
+    osThreadId_t result = osThreadNew((osThreadFunc_t)pre_main, NULL, &_main_thread_attr);
+    if ((void *)result == NULL) {
+        error("Pre main thread not created, stack memory unaligned or stack size 0?");
```

Comment:
I think we shouldn't suggest the issues as there might be many, just printing that 'Main thread not created' should be enough.